rxbilling/src/main/java/com/gen/rxbilling/client/RxBilling.kt

```diff
@@ -99,81 +98,104 @@ class RxBillingImpl(billingFactory: BillingClientFactory)
                 }
                 .firstOrError()
                 .flatMapCompletable {
-                    return@flatMapCompletable if (isSuccess(it)) {
+                    return@flatMapCompletable if (isSuccess(it.responseCode)) {
                         Completable.complete()
                     } else {
-                        Completable.error(BillingException.fromCode(it))
+                        Completable.error(BillingException.fromResult(it))
                     }
                 }
     }
 
-    override fun consumeProduct(purchaseToken: String): Completable {
+    override fun consumeProduct(params: ConsumeParams): Completable {
         return connectionFlowable
                 .flatMap { client ->
                     Flowable.create<Int>({
-                        client.consumeAsync(purchaseToken) { responseCode, _ ->
+                        client.consumeAsync(params) { result, _ ->
                             if (it.isCancelled) return@consumeAsync
+                            val responseCode = result.responseCode
                             if (isSuccess(responseCode)) {
                                 it.onNext(responseCode)
                                 it.onComplete()
                             } else {
-                                it.onError(BillingException.fromCode(responseCode))
+                                it.onError(BillingException.fromResult(result))
                             }
                         }
                     }, BackpressureStrategy.LATEST)
                 }
                 .firstOrError()
-                .toCompletable()
+                .ignoreElement()
     }
 
-    private fun getBoughtItems(type: String): Single<List<Purchase>> {
+    override fun acknowledge(params: AcknowledgePurchaseParams): Completable {
         return connectionFlowable
-                .flatMap {
-                    val purchasesResult = it.queryPurchases(type)
-                    return@flatMap if (isSuccess(purchasesResult.responseCode)) {
-                        Flowable.just(purchasesResult.purchasesList.orEmpty())
-                    } else {
-                        Flowable.error<List<Purchase>>(BillingException.fromCode(purchasesResult.responseCode))
-                    }
-                }.firstOrError()
+                .flatMap { client ->
+                    Flowable.create<Int>({
+                        client.acknowledgePurchase(params) { result ->
+                            if (it.isCancelled) return@acknowledgePurchase
+                            val responseCode = result.responseCode
+                            if (isSuccess(responseCode)) {
+                                it.onNext(responseCode)
```

Comment:
Same goes here, `responseCode` will never be received by any of the possible subscribers.